app/src/main/java/ca/pkay/rcloneexplorer/Services/SyncService.java

```diff
@@ -112,34 +117,45 @@ protected void onHandleIntent(@Nullable Intent intent) {
-                        String s = line.substring(12).trim();
-                        notificationBigText[0] = s;
-                        notificationContent = s;
-                    } else if (line.startsWith(" *")) {
-                        String s = line.substring(2).trim();
-                        notificationBigText[1] = s;
-                    } else if (line.startsWith("Errors:")) {
-                        notificationBigText[2] = line;
-                    } else if (line.startsWith("Checks:")) {
-                        notificationBigText[3] = line;
-                    } else if (line.matches("Transferred:\\s+\\d+\\s+/\\s+\\d+,\\s+\\d+%$")) {
-                        notificationBigText[4] = line;
-                    } else if (isLoggingEnable && line.startsWith("ERROR :")){
+                    JSONObject logline = new JSONObject(line);
+                    FLog.e("LOG", logline.toString());
+                    if(isLoggingEnable && logline.getString("level").equals("error")){
                         log2File.log(line);
+                    } else if(logline.getString("level").equals("warning")){
+                        //available stats:
+                        //bytes,checks,deletedDirs,deletes,elapsedTime,errors,eta,fatalError,renames,retryError
+                        //speed,totalBytes,totalChecks,totalTransfers,transferTime,transfers
+                        stats = logline.getJSONObject("stats");
+
+                        String speed = humanReadableBytes(stats.getLong("speed"))+"/s";
+                        String size = humanReadableBytes(stats.getLong("bytes"));
+                        String allsize = humanReadableBytes(stats.getLong("totalBytes"));
+                        double percent = ((double)  stats.getLong("bytes")/stats.getLong("totalBytes"))*100;
+
+                        //todo: translate
+                        notificationContent = String.format("Transfered:   %s / %s %.0f%% %s, ETA %s s",
+                                size, allsize, percent, speed, stats.get("eta"));
+                        notificationBigText[0]=notificationContent;
+                        notificationBigText[1]=String.format("Errors:      %d", stats.getInt("errors"));
+                        notificationBigText[2]=String.format("Checks:      %d / %d", stats.getInt("checks"),  stats.getInt("totalChecks"));
+                        notificationBigText[3]=String.format("Transferred: %s / %s", size, allsize);
+                        notificationBigText[4]=String.format("Elapsed:     %d", stats.getInt("elapsedTime"));
                     }
 
                     updateNotification(title, notificationContent, notificationBigText);
                 }
             } catch (IOException e) {
                 FLog.e(TAG, "onHandleIntent: error reading stdout", e);
+            } catch (JSONException e) {
+                //e.printStackTrace();
```

Comment:
Only ignore errors if they are truly benign - in this case, it may be useful to display an appropriate message to the user.
Also, use `FLog.e(TAG, "onHandleIntent: error reading json", e);` or something similar.

Reply:
done as suggested